src/lib/molecules/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,604 @@
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+import classnames from 'classnames';
+
+// Helpers
+// @ts-ignore
+import { noop } from 'utils';
+// @ts-ignore
+import { useDeviceType } from 'hooks';
+
+// Components
+import Scrollbar from '../../atoms/Scrollbar';
+// @ts-ignore
+import Icon from '../../atoms/Icon';
+// @ts-ignore
+import ExtendedInput from '../ExtendedInput';
+// @ts-ignore
+import ValidatableNumberInput from '../ValidatableElements/Elements/ValidatableNumberInput';
+import TimePickerPopover from './TimePickerPopover';
+
+// Styles
+import './TimePicker.scss';
+
+function generateTimeValues(format: string | string[], max: number) {
+    const numbers: string[] = [];
+    const formatLength = format.length;
+
+    for (let i = 0; i <= max; i++) {
+        numbers.push(i < 10 && formatLength !== 1 ? `0${i}` : String(i));
+    }
+
+    return numbers;
+}
+
+const checkFormatValidation = (format: string | string[] | undefined, value: string | undefined) =>
+    (format?.length === 1 && value?.length && value?.length <= 2 && value && Number(value[0]) !== 0) ||
+    (format?.length === 2 && value && value.length === 2);
+
+const checkHourRange = (value: string, format: string | string[]) => {
+    return value.length <= 2 && Array.isArray(format)
+        ? Number(value) < 12
+        : (isLongHour(format) && Number(value) < 24) || (isShortHour(format) && Number(value) < 12);
+};
+
+const checkRange = (value: string) => value.length <= 2 && Number(value) < 60;
+
+const isLongHour = (str: string | string[]) => str === 'HH' || str === 'H';
+const isShortHour = (str: string | string[]) => str === 'hh' || str === 'h';
+
+function convertToFormat(value: string, format: string | string[] | undefined, notEmpty: boolean = true) {
+    if (!value && notEmpty) {
+        return format?.length === 1 ? '0' : '00';
+    }
+    if (format?.length === 2 && value?.length === 1) {
+        return `0${value}`;
+    }
+    if (format?.length === 1 && value?.length === 2) {
+        return Number(value).toString();
+    }
+
+    return value;
+}
+
+interface ITimePickerProps {
+    /**
+     * Define is seconds field will be shown or no
+     */
+    showSeconds?: boolean;
+    /**
+     * Select view with multiple inputs or with single inputs <br>
+     * Possible values: `'multipleInputs' | 'singleInput'`
+     */
+    appearance?: 'multipleInputs' | 'singleInput';
+    /**
+     * Format for hour field <br>
+     * Possible values: `'HH' | 'H' | 'hh' | 'h' | ['HH', 'H', 'hh', 'h']`
+     */
+    hourFormat?: 'HH' | 'H' | 'hh' | 'h' | string[];
```

Comment:
1  the * Possible values:  cant be `['HH', 'H', 'hh', 'h']` it should be ['HH' | 'H' | 'hh' | 'h']
2. Please change the type , because the array can contain only `'HH' | 'H' | 'hh' | 'h'` so not `string[]` it should be `['HH' | 'H' | 'hh' | 'h']`.
3. this comment about `minuteFormat` and `secondFormat` to.